### PR TITLE
Refactor main.pde

### PR DESCRIPTION
**Pull request by david2544**
### We used to have a big `draw()` function that had everything in it. This was the result of combining the two animations. After the refactoring we have the following structure.

* The function `draw()` is now only used to render one of the two animations
* We have `firstScreen()` and `secondScreen()` functions for the main logic of each animation
* We have a `pixelParser()` function that is used by both animations, this code was repetitive previously.
* The `mousePressed()` function is still there for testing purposes only when we don't have a kinect nearby. However, this function is outdated and will be either updated or removed in a future PR.

## Please take a look and let me know if everything looks good from your side @AstiV @snzew .

### CODE REVIEW

File: main/main.pde [hunk 1]
```diff
@@ -18,10 +18,6 @@ final float SHRINK_RATE = 1;//2;//5;
 final int MAX_PARTICLES = 100;
 final int SPAWN_DELAY = 50; //ms
 
-int[] x1 = {320, 335, 350, 360, 372, 378, 380, 378, 372, 360, 350, 335, 320, 305, 290, 280, 272, 262, 260, 262, 268, 280, 290, 305};
-int[] y1 = {180, 182, 190, 200, 215, 228, 240, 252, 270, 280, 290, 298, 300, 298, 290, 280, 270, 256, 240, 228, 215, 200, 190, 182};
-int backgroundCounter = 10;
-
```
Comment (AstiV) [inline main/main.pde:24]: Why was this removed?

File: main/main.pde [hunk 2]
```diff
@@ -41,101 +40,102 @@ void setup()
   kinect.initDepth();
   kinect.enableMirror(true);
   background(0);
-  particleSystem.addParticle(new PVector(random(width), random(height)));
+  // particleSystem.addParticle(new PVector(random(width), random(height)));
```
Comment (AstiV) [inline main/main.pde]: Do we still need this?

File: main/main.pde [hunk 3]
```diff
@@ -41,101 +40,102 @@ void setup()
...
+  ellapsedTime = millis();
+  if(ready == true) {
+    ready = false;
+    for(int i = 0; i < 24; i++) {
+      system.addParticle(new PVector(random(x-50, x+50), random(y-50, y+50)));
+    }
+  }
+}
+
+void mousePressed() {
```
Comment (AstiV) [inline main/main.pde]: Do we need a mousePressed function here? Shouldn't we get rid of this?

**Review by AstiV — COMMENTED** [review]
Thanks for refactoring!

File: main/main.pde (hunk 1 above)
Comment (david2544) [reply main/main.pde:24]: We are now generating the particles at random points around the location where the user presses instead of having hardcoded values. You can see this on line 129
` system.addParticle(new PVector(random(x-50, x+50), random(y-50, y+50)));`

File: main/main.pde (hunk 2 above)
Comment (david2544) [reply main/main.pde]: Right! We don't, ill get rid of it.

File: main/main.pde (hunk 3 above)
Comment (david2544) [reply main/main.pde]: Removed it here: b8eb31893043e1cf7caa21fdd8bd6e695dbdea54

File: main/main.pde (hunk 2 above)
Comment (david2544) [reply main/main.pde]: Removed here: 6aec219b063e48ed8b6752364a7a71583154636b

**Review by sncis — APPROVED** [review]

File: main/main.pde [hunk 4]
```diff
@@ -41,102 +40,92 @@ void setup()
...
-  	//img.updatePixels();
-	
-	  PVector avgPosition = new PVector(avgX, avgY);
-
-    if(totalPixels > 4500){
-      particleSystem.getAttracted(avgPosition);
-    } else if (totalPixels > 0 && totalPixels < 4000){
-      particleSystem.getRepulsed(avgPosition);
+      } else {
+        // print(millis(), ellapsedTime, "\n");
```
Comment (sncis) [inline main/main.pde]: do we still need this line?

Comment (david2544) [reply main/main.pde]: Removed here: 0f5152541bc7dae94776ca4dc74e2aaec2ed724a
